src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferConvertToHexStringOverBitConverter.cs

```diff
@@ -0,0 +1,329 @@
+                {
+                    bitConverterToStringInvocation = invocation;
+                    replacementMethod = replacementToLower;
+                    // Reset toLowerInvocation as we no longer need a ToLower after using ToHexStringLower
+                    toLowerInvocation = default;
+
+                    return true;
+                }
+                else if (TryGetBitConverterReplacement(invocation.TargetMethod, out var replacement))
+                {
+                    bitConverterToStringInvocation = invocation;
+                    replacementMethod = replacement;
+
+                    return true;
+                }
+
+                return false;
+
+                static bool TryGetInstanceInvocation(IInvocationOperation invocation, [NotNullWhen(true)] out IInvocationOperation? instanceInvocation)
+                {
+                    instanceInvocation = invocation.GetInstance() as IInvocationOperation;
+
+                    return instanceInvocation is not null;
+                }
+            }
+
+            private bool IsAnyStringReplaceMethod(IMethodSymbol? method)
+            {
+                return _stringReplaceMethods.Any(m => SymbolEqualityComparer.Default.Equals(m, method));
+            }
+
+            private bool IsAnyStringToLowerMethod(IMethodSymbol? method)
+            {
+                return _stringToLowerMethods.Any(m => SymbolEqualityComparer.Default.Equals(m, method));
+            }
+
+            private bool IsStringEmptyField(IOperation operation)
+            {
+                return operation is IFieldReferenceOperation fieldReferenceOperation &&
+                    SymbolEqualityComparer.Default.Equals(fieldReferenceOperation.Field, _stringEmptyField);
+            }
+
+            private bool HasStringReplaceArgumentsToRemoveHyphen(ImmutableArray<IArgumentOperation> arguments)
+            {
+                var argumentsInParameterOrder = arguments.GetArgumentsInParameterOrder();
+                var oldValue = argumentsInParameterOrder[0].Value;
+                var newValue = argumentsInParameterOrder[1].Value;
+
+                bool oldValueIsConstantHyphenString = oldValue.ConstantValue.HasValue &&
+                    oldValue.ConstantValue.Value is string oldValueString &&
+                    oldValueString.Equals("-", StringComparison.Ordinal);
+
+                bool newValueIsConstantNullOrEmptyString = newValue.ConstantValue.HasValue &&
+                    newValue.ConstantValue.Value is string newValueString &&
+                    string.IsNullOrEmpty(newValueString);
+
+                return oldValueIsConstantHyphenString &&
+                    (newValueIsConstantNullOrEmptyString || newValue.HasNullConstantValue() || IsStringEmptyField(newValue));
+            }
+
+            private bool TryGetBitConverterReplacement(IMethodSymbol method, [NotNullWhen(true)] out IMethodSymbol? replacement)
```

Comment:
This method feels redundant, `_bitConverterReplacements` is a dictionary why we could not use its `TryGetValue` directly? `_bitConverterReplacements.TryGetValue(method, out replacement)`

Reply:
Thanks, fixed in 03110732e85f9dceebacfa846ce8e4762ab7c2df, not sure what got into me when writing this helper 😅